x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/ProcessedField.java

```diff
@@ -47,12 +51,12 @@ public ProcessedField(PreProcessor processor) {
                 continue;
             }
             final Object value = values[0];
-            if (values.length == 1 && (value instanceof String || value instanceof Number)) {
+            if (values.length == 1 && (isValidValue(value))) {
                 inputs.put(field, value);
             }
         }
         preProcessor.process(inputs);
-        return preProcessor.outputFields().stream().map(inputs::get).toArray();
+        return preProcessor.outputFields().stream().sorted().map(inputs::get).toArray();
```

Comment:
If the preprocessor already returns the output fields sorted, we shouldn't need to sort here.

Now unfortunately there is not sorted list interface. But as we're dealing with fields, if we want to enforce this to the PreProcessor interface we should switch to `SortedSet`.

Reply:
The only one that causes this problem is the Onehot processor, I will see if I can figure out a way for the order to always be the same. The order will always have to be the same after serializing from the wire and on xcontent.

Reply:
Yeah, I don't know what I was thinking 🤦 

I will fix this...The preprocessor output should be consistent